### PR TITLE
Phase 1: Foundation & Comprehensive Test Suite

### PR DESCRIPTION
**Pull request by Mohaamedl**
## 🎯 Overview
This PR implements the Phase 1 foundation for the minishell project and adds a comprehensive test suite for all built-in commands.

### Test Infrastructure
- ✅ Created master test runner (`run_all_tests.sh`) with colored output and progress tracking
- ✅ Implemented 113 comprehensive test cases across 7 test suites
- ✅ **100% test pass rate** - All 113 tests passing

### Individual Test Suites
1. **echo** (24 tests) - Basic output, `-n` flag, edge cases, special characters
2. **pwd** (12 tests) - Basic functionality, after cd operations, with arguments
3. **cd** (20 tests) - Absolute/relative paths, error handling, HOME directory
4. **env** (11 tests) - Display format, after export/unset operations
5. **export** (16 tests) - Variable validation, multiple exports, error cases
6. **unset** (13 tests) - Variable removal, special variables, validation
7. **exit** (17 tests) - Exit codes, modulo 256 wrapping, error handling

### Documentation
- Added `JIRA_STATUS.md` for project status tracking and reconciliation

## 🧪 Test Results
```
╔════════════════════════════════════════════════════════════════╗
║  🎉 🎊 ALL TEST SUITES PASSED! EXCELLENT WORK! 🎊 🎉          ║
╚════════════════════════════════════════════════════════════════╝

Total test suites run: 7
Passed: 7
Failed: 0
Success rate: 100.0%
```

## 📊 Test Coverage Details
- **echo**: 100% (24/24 tests)
- **pwd**: 100% (12/12 tests)
- **cd**: 100% (20/20 tests)
- **env**: 100% (11/11 tests)
- **export**: 100% (16/16 tests)
- **unset**: 100% (13/13 tests)
- **exit**: 100% (17/17 tests)

## 🔗 Related Issues
Resolves: KAN-60 (Create Unit Tests for Built-ins)

## ✅ Checklist
- [x] All tests passing (113/113)
- [x] Code follows 42 norminette standards
- [x] Test files have proper 42 headers
- [x] Master test runner created
- [x] Individual test files for each built-in
- [x] Comprehensive edge case coverage
- [x] Error handling validation
- [x] Documentation updated (JIRA_STATUS.md)

## 🚀 Running Tests
```bash
# Run all test suites
./tests/run_all_tests.sh

# Run individual test suites
./tests/test_echo.sh
./tests/test_cd.sh
./tests/test_pwd.sh
# ... etc
```

## 📝 Notes
- Test framework properly handles shell prompts and output filtering
- Special handlers implemented for `-n` flag tests (no newline output)
- Error message validation included for all error cases
- All test files are executable with proper permissions

### CODE REVIEW

File: src/builtins/exit.c [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+	if (arg_count > 2)
+	{
+		print_error("exit", NULL, "too many arguments");
+		return (ERROR);
+	}
+	exit_status = ft_atoi(args[1]);
+	exit_status = (unsigned char)exit_status;
+	cleanup_and_exit(shell, exit_status);
+	return (SUCCESS);
+}
```
Comment (KikoSilva1) [inline src/builtins/exit.c]: The standard ft_atoi from the libft does not handle long long overflow. If the argument is a number that exceeds the long long limit (and is valid in bash), your shell must detect it and exit with the non-numeric argument error (status 2).
